### PR TITLE
moved createCanvas to a function that i can control errors better

### PR DESCRIPTION
**Pull request by SkyJedi**

### CODE REVIEW

File: client/archonMaker.js [hunk 1]
```diff
@@ -226,12 +228,15 @@ const getCurvedFontSize = (length) => {
 
 const getCircularText = (text = '', diameter, kerning) => {
     return new Promise((resolve, reject) => {
```
Comment (cryogen) [inline client/archonMaker.js:230]: this probably doesnt need to be a promise either

File: client/archonMaker.js [hunk 2]
```diff
@@ -226,12 +228,15 @@ const getCurvedFontSize = (length) => {
...
-            reject();
+        let canvas;
+        try {
+            canvas = fabric.util.createCanvasElement();
+        } catch(err) {
+            reject(err);
             return;
         }
 
+
```
Comment (cryogen) [inline client/archonMaker.js:239]: excess blank space will likely fail linting

File: client/archonMaker.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { fabric } from 'fabric';
 import QRCode from 'qrcode';
 import uuid from 'uuid';
 
-export const buildDeckList = (deck, language, translate, AllCards) => new Promise(resolve => {
+export const buildDeckList = (deck, language, translate, AllCards) => new Promise(async resolve => {
```
Comment (cryogen) [inline client/archonMaker.js:5]: think this async can go away now

File: client/archonMaker.js [hunk 4]
```diff
@@ -171,16 +172,17 @@ export const buildDeckList = (deck, language, translate, AllCards) => new Promis
         .catch(() => resolve(defaultCard));
 });
 
-export const buildArchon = (deck, language) => new Promise(resolve => {
+export const buildArchon = (deck, language) => new Promise(async resolve => {
```
Comment (cryogen) [inline client/archonMaker.js:175]: and this one

File: client/archonMaker.js (hunk 1 above)
Comment (SkyJedi) [reply client/archonMaker.js:230]: it does until i fix another thing

Comment (SkyJedi) [reply client/archonMaker.js]: fixed the other thing

**Comment by SkyJedi** [comment]
this should be all finished now

**Pull request merged**